test/e2e/federated-service.go

```diff
@@ -148,49 +144,67 @@ var _ = framework.KubeDescribe("Service [Feature:Federation]", func() {
+	Describe("DNS", func() {
+		BeforeEach(func() {
+			framework.SkipUnlessFederated(f.Client)
+			createService(f.FederationClientset_1_3, clusterClientSets, f.Namespace.Name)
+		})
 
-		// Delete a federated service shard in the default e2e Kubernetes cluster.
-		err := f.Clientset_1_3.Core().Services(f.Namespace.Name).Delete(FederatedServiceName, &api.DeleteOptions{})
-		Expect(err).NotTo(HaveOccurred())
-		waitForFederatedServiceShard(f.Clientset_1_3, f.Namespace.Name, nil, 0)
+		It("should be able to discover a federated service", func() {
+			framework.SkipUnlessFederated(f.Client)
 
-		localSvcDNSNames := []string{
-			FederatedServiceName,
-			fmt.Sprintf("%s.%s", FederatedServiceName, f.Namespace.Name),
-			fmt.Sprintf("%s.%s.svc.cluster.local.", FederatedServiceName, f.Namespace.Name),
-		}
-		for _, name := range localSvcDNSNames {
-			discoverService(f, name, false)
-		}
+			svcDNSNames := []string{
+				FederatedServiceName,
+				fmt.Sprintf("%s.%s", FederatedServiceName, f.Namespace.Name),
+				fmt.Sprintf("%s.%s.svc.cluster.local.", FederatedServiceName, f.Namespace.Name),
+				fmt.Sprintf("%s.%s.%s", FederatedServiceName, f.Namespace.Name, federationName),
+				fmt.Sprintf("%s.%s.%s.svc.cluster.local.", FederatedServiceName, f.Namespace.Name, federationName),
+			}
+			// TODO(mml): This could be much faster.  We can launch all the test
+			// pods, perhaps in the BeforeEach, and then just poll until we get
+			// successes/failures from them all.
+			for _, name := range svcDNSNames {
+				discoverService(f, name)
+			}
+		})
 
-		svcDNSNames := []string{
-			fmt.Sprintf("%s.%s.%s", FederatedServiceName, f.Namespace.Name, federationName),
-			fmt.Sprintf("%s.%s.%s.svc.cluster.local.", FederatedServiceName, f.Namespace.Name, federationName),
-		}
-		for _, name := range svcDNSNames {
-			discoverService(f, name, true)
-		}
+		Context("non-local federated service", func() {
+			BeforeEach(func() {
+				framework.SkipUnlessFederated(f.Client)
+
+				// Delete a federated service shard in the default e2e Kubernetes cluster.
+				err := f.Clientset_1_3.Core().Services(f.Namespace.Name).Delete(FederatedServiceName, &api.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				waitForFederatedServiceShard(f.Clientset_1_3, f.Namespace.Name, nil, 0)
+			})
+
+			It("should be able to discover a non-local federated service", func() {
+				framework.SkipUnlessFederated(f.Client)
+
+				svcDNSNames := []string{
+					fmt.Sprintf("%s.%s.%s", FederatedServiceName, f.Namespace.Name, federationName),
+					fmt.Sprintf("%s.%s.%s.svc.cluster.local.", FederatedServiceName, f.Namespace.Name, federationName),
+				}
+				for _, name := range svcDNSNames {
+					discoverService(f, name)
+				}
+
+				// TODO(mml): Unclear how to make this meaningful and not terribly
+				// slow.  How long (how many minutes?) do we verify that a given DNS
+				// lookup *doesn't* work before we call it a success?  For now,
```

Comment:
Yeah, I'm happy to leave this test disabled until we've dialed down the TTL. Just wanted to explain what the solution would be :-)